### PR TITLE
Use ActiveSupport::Notifications instead of Logger

### PR DESCRIPTION
**Pull request by mikegee**

### CODE REVIEW

File: lib/sql_footprint.rb [hunk 1]
```diff
@@ -1,88 +1,64 @@
...
+        lines.each do |line|
+          f.puts line
+        end
       end
     end
 
-    def debug?
-      true
+    def exclude
+      @capture = false
```
Comment (rsnorman) [inline lib/sql_footprint.rb]: Is it common practice to use an instance variable named the same as the method it's toggling on and off? It threw me off initially that we have an instance variable named `@capture` and a method named `capture` that looks like a setter.

Comment (mikegee) [inline lib/sql_footprint.rb]: Nah, that part is super gross. I just haven't thought of something better yet.

**Comment by mikegee** [comment]
@brandonjoyce check this out! I changed the whole thing!

File: spec/sql_footprint_spec.rb [hunk 2]
```diff
@@ -6,63 +6,73 @@
...
+      # Warmup so Widget.create just makes one new flavor of SQL
+      Widget.first
+      Cog.create!
     end
 
-    it 'logs exists sql' do
-      Widget.exists? name: 'a'
-      expect(logger.logs.length).to eq 1
+    it 'logs sql' do
+      expect { Widget.create! }.to change { described_class.lines.length }.by(+1)
```
Comment (rsnorman) [inline spec/sql_footprint_spec.rb]: Did you mean to have `+1` instead of `1` here? And if you did, I'd love to learn the advantage over the latter.

Comment (mikegee) [inline spec/sql_footprint_spec.rb]: It conveys the increase more, maybe? And sometimes there are other expectations nearby with `-1`, which gives a nice symmetry.

Not very strong arguments, but that's what I was thinking...

**Comment by brandonjoyce** [comment]
Nice!  I knew there had to be a better way than I was doing it.

File: sql_footprint.gemspec [hunk 3]
```diff
@@ -9,7 +9,7 @@ Gem::Specification.new do |spec|
   spec.authors       = ['Brandon Joyce']
   spec.email         = ['bjoyce@covermymeds.com']
 
-  spec.summary       = 'Keeps your DB guy happy.'
+  spec.summary       = 'Keeps your DBA happy.'
```
Comment (rsnorman) [inline sql_footprint.gemspec]: Now if I can only eliminate "guys" from my daily conversation…

File: spec/sql_footprint_spec.rb (hunk 2 above)
Comment (rsnorman) [inline spec/sql_footprint_spec.rb]: Makes sense, that's what I thought but was curious

File: lib/sql_footprint.rb (hunk 1 above)
Comment (rsnorman) [inline lib/sql_footprint.rb]: Yeah, I can't think of anything great either… `@capture_enabled`? Meh

File: lib/sql_footprint.rb [hunk 4]
```diff
@@ -1,88 +1,64 @@
...
+      sql = sql.gsub(/\[\[.*\]\]/, '')
+      sql = strip_string_values(sql)
+      sql = strip_integer_values(sql)
+      strip_in_clause_values(sql)
     end
 
-    def strip_in_clause_values text
-      text.gsub(/\sIN\s\((.*)\)/) do |_match|
+    def strip_in_clause_values sql
+      sql.gsub(/\sIN\s\((.*)\)/) do |_match|
```
Comment (schneiderderek) [inline lib/sql_footprint.rb]: I know it's common to use `_var` when you're not using a variable, but is it common to include the `|_var|` if you're not using any of the block variables. I.e. is it legit if this is actually `sql.gsub(/\sIN\s\((.*)\)/) { ' IN (values-redacted)' }`?

File: lib/sql_footprint.rb [hunk 5]
```diff
@@ -1,88 +1,64 @@
...
+      sql.gsub(/\s\=\s([0-9]+)/) do |_match|
         ' = number-redacted'
       end
     end
 
-    def strip_string_values text
-      text.gsub(/\s\=\s\'(.*)\'/) do |_match|
+    def strip_string_values sql
+      sql.gsub(/\s\=\s\'(.*)\'/) do |_match|
         " = 'value-redacted'"
```
Comment (schneiderderek) [inline lib/sql_footprint.rb]: Should these strings (and the regexs above) be constants?

File: lib/sql_footprint.rb [hunk 6]
```diff
@@ -1,88 +1,64 @@
...
+      return unless @capture
+      @lines << strip_values(sql)
     end
 
-    def strip_values text
-      text = text.gsub(/\[\[.*\]\]/, '')
-      text = strip_string_values(text)
-      text = strip_integer_values(text)
-      strip_in_clause_values(text)
+    def strip_values sql
```
Comment (schneiderderek) [inline lib/sql_footprint.rb]: What if we pulled all of this string processing out into another class?

Comment (mikegee) [inline lib/sql_footprint.rb]: It's hard to tell from the diff, but I was trying to change as little as possible to make this work. I think that's a great idea for the next PR.

**Comment by brandonjoyce** [comment]
LGTM!

**Comment by mikegee** [comment]
@schneiderderek, you left some very good comments on this PR, but I didn't want it to get bigger than it was. Could you draft another PR with your thoughts?

**Pull request merged**